test/mpi/datatypetest.txt

```diff
@@ -0,0 +1,38 @@
+# This file is used by autogen.sh to generate multiple binary files for datatype testing.
+# Every line must have the format:
+#
+# <full source file path>:<define macros>:<space separated list of counts>:timeLimit=<seconds>:<procs>
+#
+# For every line the autogen.sh script generates multiple binaries. More specifically,
+# every line will have multiple binaries corresponding to different combinations of basic
+# datatype (a list of which is defined in datatypelist.txt) and count. Each binary will be
+# built using the additional define macros (e.g., BCAST_COMM_WORLD_ONLY OTHER_STUFF).
+# Naming of produced binaries will be in the form: <filename_i_j_k> where is the number
+# of basic datatype in datatypelist.txt, j is index of the count in the count list and
+# k is the line number of this file (this is only used to differenciate betwween binaries
+# generated with different define macros keeping the filename length reasonable).
+# NOTE: first and third fields are mandatory (i.e., there must be a valid file name and count)
+
+attr/fkeyvaltype.c::1::1
+coll/bcast.c::1 8 64 512 32768 262144::4
+coll/bcast.c:BCAST_COMM_WORLD_ONLY:1 8 64 512 32768 262144::10
+cxx/attr/fkeyvaltypex.cxx::1::1
+cxx/datatype/packsizex.cxx::1::1
+pt2pt/pingping.c::1 2 4 8 16 32 64 128 256 512 1024 2048 4096 8192 16384 32768 65536 131072 262144::2
+pt2pt/sendrecv1.c::1 8 64 512 32768 262144::4
+pt2pt/sendself.c::1 8 64 512 32768 262144::1
+rma/accfence1.c::1 8 64 512 32768 262144::4
+rma/accpscw1.c::1 8 64 512 32768 262144::4
+rma/epochtest.c::1 8 64 512 32768 262144::4
+rma/getfence1.c::1 8 64 512 32768 262144::2
+rma/lock_contention_dt.c::1 8 64 512 32768 262144::4
+rma/lock_dt.c::1 8 64 512 32768 262144::2
+rma/lock_dt_flush.c::1 8 64 512 32768 262144::2
+rma/lock_dt_flushlocal.c::1 8 64 512 32768 262144::2
+rma/lockall_dt.c::1 8 64 512 32768 262144::4
+rma/lockall_dt_flush.c::1 8 64 512 32768 262144::4
+rma/lockall_dt_flushall.c::1 8 64 512 32768 262144::4
+rma/lockall_dt_flushlocal.c::1 8 64 512 32768 262144::4
+rma/lockall_dt_flushlocalall.c::1 8 64 512 32768 262144::4
+rma/putfence1.c::1 8 64 512 32768 262144::2
+rma/putpscw1.c::1 8 64 512 32768 262144::4
```

Comment:
@pavanbalaji should we also have the envvar with number of objects to test defined here and added to the `testlist.in`? It will be then responsibility of the Jenkins worker to set it for each test by looking at the final testlist

Reply:
The intent is that the jenkins script will set different environment variables in different configurations, e.g., in the PR testing we'll probably just do some basic testing, in the nightly tests we'll do more comprehensive tests, and in the per-release tests we'll do the full tests.  So it's better to not have this hardcoded in the above file.